server/errors.go

```diff
@@ -0,0 +1,33 @@
+package server
+
+import (
+	"fmt"
+
+	"github.com/Layr-Labs/eigenda-proxy/commitments"
+)
+
+// HandlerError is an interface that can be either a simple error or a MetaError
+type HandlerError interface {
+	error
+	isHandlerError() // marker method
+}
```

Comment:
Why are you defining an interface here? Don't think we should be doing this. We want to adhere to the error interface that golang defines, and then we just define custom errors by defining structs that implement the Error() function, and that's it. Then we have access to https://pkg.go.dev/errors#Is and a bunch of goodies.